### PR TITLE
Detect Vulkan Xlib/Wayland surface support at runtime

### PR DESCRIPTION
**Pull request by myfreeweb**

### CODE REVIEW

**Comment by hcorion** [comment]
Confirming that this fixed the issue for me.

File: rpcs3/Emu/RSX/VK/VKHelpers.h [hunk 1]
```diff
@@ -1135,6 +1136,29 @@ namespace vk
 		}
 	};
 
+	class supported_extentions
```
Comment (TGEnigma) [inline rpcs3/Emu/RSX/VK/VKHelpers.h]: You typo'd extensions

Comment (myfreeweb) [reply rpcs3/Emu/RSX/VK/VKHelpers.h]: lol oops :D

**Comment by kd-11** [comment]
```
../rpcs3/Emu/RSX/VK/VKHelpers.h:1258:29: error: ‘VK_KHR_WAYLAND_SURFACE_EXTENSION_NAME’ was not declared in this scope
    if (support.is_supported(VK_KHR_WAYLAND_SURFACE_EXTENSION_NAME)) {
```

**Comment by myfreeweb** [comment]
huh, I guess that needs the ifdef still

**Comment by kd-11** [comment]
Looks like not vulkan headers have the macro definition for wayland surface extension. You can manually link to an external deb which has said declaration and modify travis.ci or use the extension name itself. Travis isn't building as is, making this unmergable. I'd like to get this in fast as it is a hotfix.

**Comment by kd-11** [comment]
Disregard last, found it
```
#ifdef VK_USE_PLATFORM_WAYLAND_KHR
#define VK_KHR_wayland_surface 1
#include <wayland-client.h>

#define VK_KHR_WAYLAND_SURFACE_SPEC_VERSION 6
#define VK_KHR_WAYLAND_SURFACE_EXTENSION_NAME "VK_KHR_wayland_surface"
```

looks like a missing #ifdef like you posited

File: rpcs3/Emu/RSX/VK/VKHelpers.h [hunk 2]
```diff
@@ -1135,6 +1136,29 @@ namespace vk
...
+	private:
+		std::vector<VkExtensionProperties> m_vk_exts;
+
+	public:
+
+		supported_extensions()
+		{
+			uint32_t count;
+			if (vkEnumerateInstanceExtensionProperties(nullptr, &count, nullptr) != VK_SUCCESS)
+				return;
```
Comment (kd-11) [inline rpcs3/Emu/RSX/VK/VKHelpers.h]: add space after the return here

File: rpcs3/Emu/RSX/VK/VKHelpers.h [hunk 3]
```diff
@@ -1216,21 +1240,32 @@ namespace vk
...
+			extensions.push_back(VK_EXT_DEBUG_REPORT_EXTENSION_NAME);
 #ifdef _WIN32
-				VK_KHR_WIN32_SURFACE_EXTENSION_NAME,
+			extensions.push_back(VK_KHR_WIN32_SURFACE_EXTENSION_NAME);
 #else
-				VK_KHR_XLIB_SURFACE_EXTENSION_NAME,
-#endif
+			supported_extensions support;
+			bool found_surface_ext = false;
+			if (support.is_supported(VK_KHR_XLIB_SURFACE_EXTENSION_NAME)) {
```
Comment (kd-11) [inline rpcs3/Emu/RSX/VK/VKHelpers.h]: brace on newline

File: rpcs3/Emu/RSX/VK/VKHelpers.h [hunk 4]
```diff
@@ -1216,21 +1240,32 @@ namespace vk
...
+				extensions.push_back(VK_KHR_XLIB_SURFACE_EXTENSION_NAME);
+				found_surface_ext = true;
+			}
 #ifdef VK_USE_PLATFORM_WAYLAND_KHR
-				VK_KHR_WAYLAND_SURFACE_EXTENSION_NAME,
+			if (support.is_supported(VK_KHR_WAYLAND_SURFACE_EXTENSION_NAME)) {
```
Comment (kd-11) [inline rpcs3/Emu/RSX/VK/VKHelpers.h]: same here

File: rpcs3/Emu/RSX/VK/VKHelpers.h [hunk 5]
```diff
@@ -1216,21 +1240,32 @@ namespace vk
...
+				extensions.push_back(VK_KHR_WAYLAND_SURFACE_EXTENSION_NAME);
+				found_surface_ext = true;
+			}
+#endif
+			if (!found_surface_ext) {
```
Comment (kd-11) [inline rpcs3/Emu/RSX/VK/VKHelpers.h]: and here

**Pull request merged**